tests/std/tests/P1518R2_stop_overconstraining_allocators/test.compile.pass.cpp

```diff
@@ -0,0 +1,131 @@
+    [[maybe_unused]] auto y = SfinaeTester<list>::test(move(px), &mr);
+}
+
+void test_vector(pmr::vector<int>& px) {
+    [[maybe_unused]] auto x = SfinaeTester<vector>::test(px, &mr);
+    [[maybe_unused]] auto y = SfinaeTester<vector>::test(move(px), &mr);
+}
+
+void test_vector_bool(pmr::vector<bool>& px) {
+    [[maybe_unused]] auto x = SfinaeTester<vector>::test(px, &mr);
+    [[maybe_unused]] auto y = SfinaeTester<vector>::test(move(px), &mr);
+}
+
+void test_map(pmr::map<int, int>& px) {
+    [[maybe_unused]] auto x = SfinaeTester<map>::test(px, &mr);
+    [[maybe_unused]] auto y = SfinaeTester<map>::test(px, &mr);
+}
+
+void test_multimap(pmr::multimap<int, int>& px) {
+    [[maybe_unused]] auto x = SfinaeTester<multimap>::test(px, &mr);
+    [[maybe_unused]] auto y = SfinaeTester<multimap>::test(move(px), &mr);
+}
+
+void test_multiset(pmr::multiset<int>& px) {
+    [[maybe_unused]] auto x = SfinaeTester<multiset>::test(px, &mr);
+    [[maybe_unused]] auto y = SfinaeTester<multiset>::test(move(px), &mr);
+}
+
+void test_set(pmr::set<int>& px) {
+    [[maybe_unused]] auto x = SfinaeTester<set>::test(px, &mr);
+    [[maybe_unused]] auto y = SfinaeTester<set>::test(move(px), &mr);
+}
+
+void test_unordered_map(pmr::unordered_map<int, int>& px) {
+    [[maybe_unused]] auto x = SfinaeTester<unordered_map>::test(px, &mr);
+    [[maybe_unused]] auto y = SfinaeTester<unordered_map>::test(move(px), &mr);
+}
+
+void test_unordered_multimap(pmr::unordered_multimap<int, int>& px) {
+    [[maybe_unused]] auto x = SfinaeTester<unordered_multimap>::test(px, &mr);
+    [[maybe_unused]] auto y = SfinaeTester<unordered_multimap>::test(move(px), &mr);
+}
+
+void test_unordered_multiset(pmr::unordered_multiset<int>& px) {
+    [[maybe_unused]] auto x = SfinaeTester<unordered_multiset>::test(px, &mr);
+    [[maybe_unused]] auto y = SfinaeTester<unordered_multiset>::test(move(px), &mr);
+}
+
+void test_unordered_set(pmr::unordered_set<int>& px) {
+    [[maybe_unused]] auto x = SfinaeTester<unordered_set>::test(px, &mr);
+    [[maybe_unused]] auto y = SfinaeTester<unordered_set>::test(move(px), &mr);
+}
+
+void test_priority_queue1(priority_queue<int, pmr::vector<int>>& px) {
+    [[maybe_unused]] auto x = SfinaeTester<priority_queue>::test(px, &mr);
+    [[maybe_unused]] auto y = SfinaeTester<priority_queue>::test(move(px), &mr);
+}
+
+void test_queue1(queue<int, pmr::deque<int>>& px) {
+    [[maybe_unused]] auto x = SfinaeTester<queue>::test(px, &mr);
+    [[maybe_unused]] auto y = SfinaeTester<queue>::test(move(px), &mr);
+}
+
+void test_stack1(stack<int, pmr::vector<int>>& px) {
+    [[maybe_unused]] auto x = SfinaeTester<stack>::test(px, &mr);
+    [[maybe_unused]] auto y = SfinaeTester<stack>::test(move(px), &mr);
+}
+
+void test_priority_queue2(less<int> comp, pmr::vector<int>& pc) {
+    [[maybe_unused]] auto x = SfinaeTester<priority_queue>::test(comp, pc, &mr);
+    [[maybe_unused]] auto y = SfinaeTester<priority_queue>::test(comp, move(pc), &mr);

```

Comment:
We additionally need a `test_priority_queue3` to cover the constructors that correspond to the guide `priority_queue(_Iter, _Iter, _Compare, _Container, _Alloc)`.